### ISSUE TITLE
Miscellaneous editor feedback and usability issues

### ISSUE THREAD

**Issue by lewislepton** [AtomicGameEngine/AtomicGameEngine#159]
so. rather than post everything in gitter for suggestions and not put it in my own list of wishes - this is better. these are just my own thoughts about what i would like to see. im not begging for these, these are just my views on what would be cool for atomic.
these are either cosmetic or actual features of use
i am using the repo version also, if you are wondering ;)

• after running a project. a window pops up saying what has happened. this should go into console instead.
• key layouts other than QWERTY need to be addressed. such as the camera/scene movement. you can move the camera with WASD, but for others without a QWERTY layout, this is problematic. arrow keys instead?
• to be able to minimive/maximize views, such as the sidebars and move things to your liking
• saves your layout
• to be able to make text in the editor bigger/smaller
• move the top bar 'play' options away and put them either next to the actual top bar, next to file-edit etc. or above the 'project' name on the top left, or put it to the top right above 'inspector'. this leaves room for the tabs to move up and take that space, giving more for scene view
• take away the cartoony tick boxes in the inspector. they are out of sync with atomic. i would go for a blue/color box that lights when on, grey when off. a bit like the dropdown markers in inspector
• also the cartoony 'x' for closing files are out of place. plus are a little hard to see. 

i had some more. but really thats good for now.
ill add as and when i think of them. but something just thats on my mind when i first use atomic

**Comment by lewislepton** [comment]
• `Atomic.Animation2D` & `Atomic.Animation3D`. rather than just `Atomic.Animation3D` [wihch it currently is] or the other. i think that helps as well, just to be sure people understand its a 2d or 3d object you are making

**Comment by lewislepton** [comment]
• grid view & grid snap in the scene view
• will tiled be used in future? it is useful and a very good tool. but a built in one much like tiled would be amazing. so rather than going from software to software. then it would be more integrated with things such as live saving/loading

**Comment by lewislepton** [comment]
• 'create component' dropdowns to expand on hover

**Comment by lewislepton** [comment]
• project name to show
• have the 'file-edit-build' etc in the top most tool bar

**Comment by lewislepton** [comment]
• XYZ co-ordinates are not updating when moving objects
![cameraupdate](https://cloud.githubusercontent.com/assets/11299322/9194651/b5ee71ea-4014-11e5-9db6-5dabdc1c1d5f.gif)

**Comment by lewislepton** [comment]
• the 'player' could be embedded into the main scene window. more a preference/choice

**Comment by lewislepton** [comment]
• not to package tiled with atomic. purely for space/download. people con grad tiled, then have a preference to locate it on your HD. which then saves

**Comment by lewislepton** [comment]
• tiled naming objects does not transend onto atomic. so a saved tiled file, with Playerspawn, do not appear in atomic. so everything is 'Object'

<img width="499" alt="screen shot 2015-08-11 at 10 27 17" src="https://cloud.githubusercontent.com/assets/11299322/9196015/3016db4c-401f-11e5-8541-71d55e0f9651.png">

**Comment by lewislepton** [comment]
• saving changed file from tiled linked with atomic project crashes. didnt pop up with crash report

**Comment by JoshEngebretson** [comment]
Lots of great notes, will break some of these out to their own issues.

**Comment by lewislepton** [comment]
cool spot. thanks

**Comment by lewislepton** [comment]
• since spriter is linked with atomic. double clicking on a .scml should open it. you would need to have a setting to locate it though in a preference

**Comment by lewislepton** [comment]
• update the icon to a retina type one. i know people will bitch. im not fussed. but you know 'they' will ;)

**Comment by lewislepton** [comment]
• when shifting around from atom to atomic. cmd-p does not start the player right away. it takes a few seconds for me to start the player. some sort of lag or something

**Comment by lewislepton** [comment]
• it would be good if the inspector dropdowns could remember states of being open or closed

**Comment by lewislepton** [comment]
• the 'create component' in inspector should put 3d items in a 3d dropdow, much like the 2d ones.. keep things ordered just to not confuse people when needing to select

**Comment by lewislepton** [comment]
• renaming created scene, does not keep name when restarting atomic

**Comment by lewislepton** [comment]
• mini camera preview window. so editgn your camera can show a brief overview of what it should look like when playing

**Comment by lewislepton** [comment]
• 'open last session' option

**Comment by lewislepton** [comment]
• some things are being cut off. maybe even have writing on the left, and UI on the right? or maybe above
<img width="249" alt="screen shot 2015-08-12 at 21 22 44" src="https://cloud.githubusercontent.com/assets/11299322/9235718/6cb143fc-4138-11e5-80ee-f35ce2f30442.png">

**Comment by lewislepton** [comment]
• if 'player output' window is not closed before starting new run. it doubles the 'player output' window

**Comment by lewislepton** [comment]
• tabbing through setting in inspector [or anywhere] should allow the entire box to be seleceted to replace. currently it only puts the cursor in to add things

**Comment by lewislepton** [comment]
• have physics indicator in the scene. to know that you have a physics object working, also to show it being edited

**Comment by lewislepton** [comment]
• if the player is within the atomic scene view. this would be great for seeing updates on co-ordinates as well

**Comment by lewislepton** [comment]
• closing tab of non active scene, makes current active scene, inactive. gif given

![aug-13-2015 13-53-29](https://cloud.githubusercontent.com/assets/11299322/9250187/f53d8eda-41c2-11e5-83c6-bcc084c8a310.gif)

**Comment by lewislepton** [comment]
• editor needs indication to show that project has not been saved

**Comment by lewislepton** [comment]
• 2d physics *needs* to have 'set from model`

**Comment by lewislepton** [comment]
• renaming node name for a scene does not save name

**Comment by lewislepton** [comment]
• pressing enter once changing value in inspector value should take away the cursor, otherwise it looks like it hasnt be saved

**Comment by lewislepton** [comment]
• pressing `ctrl - click` [mac] does not allow right click actions

**Comment by lewislepton** [comment]
• reset recent projects list button. just so we can clear it if need be

**Comment by lewislepton** [comment]
• when selecting to open a new project before closing the current one. a pop up window should ask if you

**Comment by shaddockh** [comment]
Adding a couple more cosmetic issues:
- Title bar of the editor should display which project you are currently editing
- When opening the editor, the 'play' button should be disabled since you don't actually have a project open

(running on OSX)

**Comment by lewislepton** [comment]
@shaddockh actually got the project name one up. but like the disable button if no project active

**Comment by lewislepton** [comment]
• play movie clips? for cutscenes or something similar

**Comment by lewislepton** [comment]
• pressing enter should constitute the value finished with when entering data. such as value boxes and renaming assets. instead of pressing 'ok'

**Comment by lewislepton** [comment]
• `esc` key to cancel out of things

**Comment by lewislepton** [comment]
• right click import command on folders for importing sprites/scenes/audio etc

**Comment by lewislepton** [comment]
• change of inspector name needs to change and keep name that then appears in hierarchy

**Comment by lewislepton** [comment]
• newly created scene in a 2d template spawn `PhysicsWorld' on the scene and not `PhysicsWorld2D`

**Comment by lewislepton** [comment]
• blank object creation. so you can group things. like walls etc. or even have a `group` selection option

**Comment by lewislepton** [comment]
• have 2d & 3d view with scenes. click a toggle that allows 2d or 3d view of the scene. just so you are not locked to just one. controls to move around should then change depending on setting. plus would make it easier to switch

**Comment by lewislepton** [comment]
• empty project being 2d or 3d should be empty, no sprites, models or anything. the settings in inspector and with the project should be set to either 2d or 3d. other wise we spend time getting rid of things rather than just getting on with making

**Comment by lewislepton** [comment]
• deleting object in scene does not delete the XYZ lines
![deleteme](https://cloud.githubusercontent.com/assets/11299322/9279324/a79953ca-42ae-11e5-92e0-5022d8eb8ee7.gif)

**Comment by lewislepton** [comment]
• the movement speed using WASD could be slowed down or even editable to someones liking

**Comment by lewislepton** [comment]
• click & drag to re-arrange inspector tabs and also hierarchy objects

**Comment by lewislepton** [comment]
• shortcuts dont seem to work right away from moving from windows. or when selecting the atomic window. there seems to be some lag before you can use them.
shortcuts would be, `run`, `copy & paste`, `cut`, `save` etc

**Comment by lewislepton** [comment]
• shift select multiple objects in scene so they can be moved at once

**Comment by lewislepton** [comment]
• think this is on an issue - controllers dont seem to be working right. it shows its connected, but buttons, axis etc are not working

**Comment by lewislepton** [comment]
• to have an editable background color.

**Comment by lewislepton** [comment]
• computer fan seems to kick up a fuss if atomic is sitting playing something after a minute or 2. even with simple setup such as a few rigidbodies that are standing still. might just be optimisation

**Comment by lewislepton** [comment]
• spriter animation is great [i bought spriter many moons ago ;)], but a built in small scale 2d animator that is non bone based, where you could see frames would be great. if you have ever used software like 'pyxel edit' or even 'pixen'. that type

**Comment by lewislepton** [comment]
• be able to 'lock' assets in the hierarcy. so as to stop from moving them etc

**Comment by lewislepton** [comment]
• have a remember state. so when you open your project again, it remembers your zooming from view in the scene, also if you had a scene, script etc open

**Comment by lewislepton** [comment]
• make prefabs editable from the folder rather than having to drag them into scene. have a button that can commit to those changes just incase

**Comment by lewislepton** [comment]
• slow down occuring when creating multiple rigidbody2d objects and interacting with them. only become regular once apart.
game speed - 60fps with vsync on

![slowdownbodies](https://cloud.githubusercontent.com/assets/11299322/9290065/4137ed16-437c-11e5-949a-186f6b545ac3.gif)

**Comment by lewislepton** [comment]
-continuation of previous bodies post-
it seem that it is happens most when the bodies are moving. another gif

**Comment by lewislepton** [comment]
• have a 'create script' button in the JSComponent in inspector

**Comment by lewislepton** [comment]
-continuation of previous physics post-
• this is with `continuos physics` and `sub stepping` turned off. new gif

![slowdownthing](https://cloud.githubusercontent.com/assets/11299322/9290987/28facf7a-439f-11e5-952a-4300c0ea310a.gif)

screenshot of setting in `physicsWorld2D`
<img width="319" alt="screen shot 2015-08-15 at 22 44 02" src="https://cloud.githubusercontent.com/assets/11299322/9290988/37552f8e-439f-11e5-9795-6d3a4a5e1617.png">

**Comment by lewislepton** [comment]
• latest update seems to be kicking the fan in without anything running in the player, its just stationary. which is strange

**Comment by lewislepton** [comment]
• instead of having the sprite sheet animator in atomic [as i said in a previous post]. have atomic be able to read spritesheets, so you can simply code in frames, width & height of spritesheet, but also allow to name animation frames.

this is 1 such spritesheet animation capable plugin for openf/haxe i was thinking on as an idea, and have used.
https://github.com/jgranick/spritesheet

this would be instead of having a gui based editor, wihch i think would be good, but also memory hungry. code is better ;)

**Comment by lewislepton** [comment]
• if using a text editor to edit atomic code, whether JS/TS/HAXE. when saving/building from text editor, atomic editor needs to do a mild 'reload' to show if things such as boxs, tick boxs or sound sources have been added for use. with things like `SoundSource` etc

**Comment by lewislepton** [comment]
• pex particle emittor2d loading does not seem to be working

![not loading pex](https://cloud.githubusercontent.com/assets/11299322/9305781/f31cff10-44eb-11e5-8801-9784231f8772.gif)

**Comment by lewislepton** [comment]
• yeah. atomic needs a state reload for when saving files. rather than closing then opening a scene to get entries [such as `inspectorFields`] to show up on JSComponent

**Comment by lewislepton** [comment]
• this is more a request than anything. but please oh please oh please do NOT make an asset store. i think these things are pretty pointless and actually hindering creativity.

**Comment by lewislepton** [comment]
• once clicking on atomic to start, the cursor then centres itself on screen.

**Comment by lewislepton** [comment]
can not change scale on a prefab. once changed, saved etc. it does not save the scale change

**Comment by lewislepton** [comment]
• editor should have system where it shows what is being moved, or played with. ind of like a visual update. will see if X/Y/Z or sprite flips etc are happening. this would only really work if player were to be embedded in the editor

**Comment by lewislepton** [comment]
• once asset is selected with in editor. use the arrows keys to move assets

**Comment by lewislepton** [comment]
• keyboard shortcuts for menus - these are for mac, but obviously do for others
`cmd-n` for new project - `cmd-shift-w` for close project - `cmd-o` for open project - `cmd-shift-s` save all files etc

**Comment by lewislepton** [comment]
• either the tick box in the inspector for node should enable/disable everything [including box2d stuff etc] in that node, or to have another tick box that can do that
<img width="316" alt="screen shot 2015-08-21 at 15 01 48" src="https://cloud.githubusercontent.com/assets/11299322/9409942/94fb19b2-4815-11e5-8f5e-452dbc232f70.png">

**Comment by lewislepton** [comment]
• have undo/redo in scene view

**Comment by lewislepton** [comment]
• using the player seems to center the cursor on screen

**Comment by lewislepton** [comment]
• to expand on compatibility with animation and animating.
have a xml reader for pyxel edit.
so while we have spriter reader [which is awesome], but is for bone based animations. pyxel edit could be for 2d pixel frame animations. it can export xml. here is a file

**Comment by lewislepton** [comment]
• physics collision needs fixing. doesnt work 100% the time.
say this is my player
```
node.subscribeToEvent('PhysicsBeginContact2d', onBeginContact);
private function onBeginContact(event):Void {
    if (event.nodeB != node) return;
    if (event.nodeA.name == 'DeathPit') {
      trace('TIHS IS  THE DEATH OF PLAYER');
      // node.remove();
    }
  }
```
this doesnt work with my named `DeathPit`

**Comment by lewislepton** [comment]
• profiler is cool. would be better in its own window. say like the console etc is laid out
<img width="349" alt="profiler" src="https://cloud.githubusercontent.com/assets/11299322/9429556/1162718c-49cc-11e5-89e4-ff466454e676.png">

**Comment by lewislepton** [comment]
• collected node, with other assets. such as platforms. do not change with the XYZ co-ordinates. the screenshot shows node group of platforms. the color XYZ is actually the center of the screen, but the platformMain shows 0,0,0 in position, which is wrong

<img width="852" alt="screen shot 2015-08-23 at 21 38 37" src="https://cloud.githubusercontent.com/assets/11299322/9430097/82388bea-49df-11e5-8944-3396e4e41b2a.png">

**Comment by lewislepton** [comment]
• have an additional measurement for the XYZ, such as pixels, or units etc. so can switch between and then get pixel based sprites to measure properly.

**Comment by lewislepton** [comment]
• cant copy/paste grouped nodes. though you can duplicate. though there is no keyboard shortcut for duplicate

**Comment by lewislepton** [comment]
• have a preview of sprites

**Comment by lewislepton** [comment]
• have pixel size for XYZ with things like `collisionbox2D`, plus anything else. so you could just put 32x128 instead. could be exact

**Comment by lewislepton** [comment]
• when dragging in assets/sprites in a 2d project. make the Z co-ordinate be always at `0` on import

**Comment by lewislepton** [comment]
• clicking and dragging with a full hierarchy, moves the entire list. makes it hard to put files into a blank node
![clickdrag](https://cloud.githubusercontent.com/assets/11299322/9430560/c100e206-49f2-11e5-88aa-6a088a0bb552.gif)

**Comment by lewislepton** [comment]
• `CollisionBox2D` has to overlap slightly on boxed items. if not, they have this speed bump affect when rolling over. if player is fully a square, it does not go over at slower speeds
<img width="204" alt="screen shot 2015-08-24 at 12 33 59" src="https://cloud.githubusercontent.com/assets/11299322/9439182/a20643c2-4a5c-11e5-9489-4da2a99b631b.png">

**Comment by lewislepton** [comment]
• 2 or more collision event do not work with this type of code
```
node.subscribeToEvent('PhysicsBeginContact2d', onBeginContact);
private function onBeginContact(event):Void {
    if (event.nodeB != node) return;
    if (event.nodeA.name == 'DeathPit') {
      trace('TIHS IS  THE DEATH OF PLAYER');
      // node.remove();
    }
  }
```

**Comment by lewislepton** [comment]
• .png/scene selection needs a 'back' button

**Comment by lewislepton** [comment]
• this may be the sprite renderer, but it shows black lines around curved objects. though they do show as clean when exported from the graphic application. supplied the screenshot, and the actual sprite

![brickwallleft](https://cloud.githubusercontent.com/assets/11299322/9504498/d41fbe04-4c34-11e5-939b-247f56810a0f.png)


<img width="387" alt="screen shot 2015-08-26 at 20 40 18" src="https://cloud.githubusercontent.com/assets/11299322/9504482/c5395f9e-4c34-11e5-9e69-b29876e37616.png">

**Comment by lewislepton** [comment]
• very strange thing happening with bouncing sprites. made a test. with 2 exact sprites [except for color], same script etc. 2 more sprite below [also exact], same scripts with are 'trampolines'.
on running. one does not bounce. i stop, swap the bouncing sprites around. the same one still does not bounce. but when i copy the one that doesnt bounce, then delete the original, it then works. this must be a bug

@JoshEngebretson ill send you the file since github does not deal in .zip files

**Comment by lewislepton** [comment]
• not able to copy paste grouped nodes with keyboard shortcuts

**Comment by lewislepton** [comment]
• stick fps & vsync and any other useful settings in the inspector when the scene is selected

<img width="324" alt="screen shot 2015-08-28 at 18 27 32" src="https://cloud.githubusercontent.com/assets/11299322/9552789/92f7e102-4db2-11e5-8799-6f52e1a16f53.png">

**Comment by lewislepton** [comment]
• a `purge` button.
so at the end of a project or whatever, or if you need to clean up quickly. if there are files/etc that are not being used through-out the project, this will purge them from the project file. freeing up space etc and showing all of what you are using

**Comment by lewislepton** [comment]
• include `setEmitting();` for 2d particles. rather than using `remove();`

**Comment by lewislepton** [comment]
• have an 'import sprites' option. i know we can drag and drop, but just more for options than anything

**Comment by lewislepton** [comment]
• 'Fixed Rotation' needs looking at in `RigidBody2D` in the inspector. even if on or off, it does not let a square object rotate if its hanging and falls off the edge. if off, the object should necessarily rotate if the object has reached a point
this gif shows the object being pushed set to `off` in fixed rotation

![fixedrotation](https://cloud.githubusercontent.com/assets/11299322/9584002/731f655c-5006-11e5-86bd-81ca94cb8f04.gif)

**Comment by lewislepton** [comment]
• show the size of the sprite. in a simple text box or faded text

<img width="313" alt="screen shot 2015-09-02 at 20 12 26" src="https://cloud.githubusercontent.com/assets/11299322/9641518/09b2a7f0-51af-11e5-9791-d1d980e48890.png">

**Comment by lewislepton** [comment]
• opening atomic. hold alt then press enter. it makes the screen black. have no idea what it is. but something wrong

**Comment by lewislepton** [comment]
• [ON MAC] when minimized the atomic window to dock [not hiding], then focussing on a different window. when you then maximize the atomic window, it actually resizes the window to be smaller

**Comment by lewislepton** [comment]
• it would be good if when you cycle through the hierachy with arrow keys, that it actually then selects it

**Comment by lewislepton** [comment]
• bug with activations in nodes. things are not either listened to if on or off. only when taken away are they 'off'. example given in screenshots

this is with `zone` enabled in scene
<img width="399" alt="screen shot 2015-09-06 at 17 59 28" src="https://cloud.githubusercontent.com/assets/11299322/9705584/1e9fcf84-54c1-11e5-8997-dc242936b79a.png">

this is with `zone` disabled
<img width="429" alt="screen shot 2015-09-06 at 17 59 39" src="https://cloud.githubusercontent.com/assets/11299322/9705588/28563af4-54c1-11e5-9d64-01891cbaab0b.png">

this is with `zone` deleted from scene
<img width="524" alt="screen shot 2015-09-06 at 17 59 52" src="https://cloud.githubusercontent.com/assets/11299322/9705590/33a73fb6-54c1-11e5-83e0-be47dd1e897c.png">

this also happens with other nodes, not just scene. so the enable/disable is not 100% it what is activated or deactivated

**Comment by lewislepton** [comment]
• have a color selection graph instead of numbers. more intuitive

not this
<img width="326" alt="screen shot 2015-09-06 at 22 30 02" src="https://cloud.githubusercontent.com/assets/11299322/9706729/fa14a01a-54e6-11e5-9aca-f06668d2b0cc.png">

but this
<img width="351" alt="screen shot 2015-09-06 at 22 30 42" src="https://cloud.githubusercontent.com/assets/11299322/9706730/fe216b2a-54e6-11e5-9a56-b31d91e698c1.png">

**Comment by lewislepton** [comment]
• number boxes need to have an increment available. rather than always going down/up by `1.0`. maybe hold shift while clicking to go down by `0.10`
<img width="313" alt="screen shot 2015-09-06 at 22 34 30" src="https://cloud.githubusercontent.com/assets/11299322/9706750/962a0062-54e7-11e5-93ac-e79382ed52c9.png">

**Comment by lewislepton** [comment]
• it would be good if we could set the edges of the sprites to be affected by light. if a sprite is a say 32x32, but the draw elements are smaller [because of walking animations] - it may be 30x32 for some parts, it counts the dead areas that are not being used

<img width="162" alt="screen shot 2015-09-07 at 22 40 19" src="https://cloud.githubusercontent.com/assets/11299322/9723069/afcc101c-55b1-11e5-946c-5d9f2bf60eec.png">

**Comment by lewislepton** [comment]
• to extend the color selection i mentioned before.
have HEX color input in the inspector, which also has an alpha slider

**Comment by lewislepton** [comment]
• show a radius indicator on lights in the editor

<img width="146" alt="screen shot 2015-09-08 at 17 16 12" src="https://cloud.githubusercontent.com/assets/11299322/9740785/6a593b5c-564e-11e5-9b35-2473df957a01.png">

**Comment by lewislepton** [comment]
• allow 2 different downloads. 1 which has examples built in, 1 without - just being the editor

**Comment by lewislepton** [comment]
• using 'save' on a prefab does not always save what has changed or been dadded, things like sizes, color changes etc. sometimes does not even save extra components placed inside. hard to catch it in a gif ;)

**Comment by lewislepton** [comment]
• continuing from the previous post. caught a bug in the changes made to a prefab. check the ALPHA of the pointlight2d. it changes everytime i press save, though i have not changed anything. it varies between `0.5` & `0.4`. i tried changing these to those values beforehand, but it never saved them and sometimes just deleted the pointlight

![dodgychanges](https://cloud.githubusercontent.com/assets/11299322/9884328/0c78abce-5bd7-11e5-8384-d6bd39602650.gif)

**Comment by lewislepton** [comment]
• `DirectionalLight2D` doesnt seem to work at all

**Comment by JoshEngebretson** [comment]
Ok, I have some model import functionality to get to today, but really need to start combing through here.  Thanks again, great stuff!

**Comment by lewislepton** [comment]
HA. no worries @JoshEngebretson - im actually cracking on with a project since no internet, but some things pop up that i need to tell about ;)

**Comment by lewislepton** [comment]
• a loop toggle with `SoundSource`

**Comment by lewislepton** [comment]
• when clicking on the `create component` - then if you press the spacebar, it seems to keep doubling the pop up

![doubling](https://cloud.githubusercontent.com/assets/11299322/9945791/54ada5dc-5d87-11e5-980b-35fa73e45287.gif)

**Comment by lewislepton** [comment]
• need a much better key binding way to move left right, up, down and zoom in and out in 2d. using mouse is ok, but not good if you are doing 2d

**Comment by lewislepton** [comment]
• make a new light that is like a pipe. so instead of just a point, this one can be long. so `pipe light`. just more option in lighting ;)

**Comment by lewislepton** [comment]
• sprites need an editable anchor point. so you can change the point in which it rotates. also good for placing on screen

<img width="232" alt="screen shot 2015-09-22 at 16 03 59" src="https://cloud.githubusercontent.com/assets/11299322/10022255/b30ebc5a-6143-11e5-97e7-c3a74ffe6331.png">

**Comment by lewislepton** [comment]
• snap to grid position. you can set maybe even set if its snaps by 32/64x32/64 etc, whatever you want. so it allows to place things much easier. would be best in the 'scene' inspector

<img width="339" alt="screen shot 2015-09-22 at 16 16 24" src="https://cloud.githubusercontent.com/assets/11299322/10022560/6a5b80ea-6145-11e5-863c-88735a35afb9.png">

**Comment by lewislepton** [comment]
• changing the window size does not center the window when playing it
try `Atomic.graphics.setWindowSize(400, 400);` and it does not appear in the center when playing

**Comment by lewislepton** [comment]
• naming IOS app [using the sanpshot] does not export with its  given name
![e](https://cloud.githubusercontent.com/assets/11299322/10057185/c9d9cd64-6236-11e5-9ee5-8776019e6ac2.png)

**Comment by lewislepton** [comment]
• running on ipad, it makes it application like iphone, since it shows the 1x buton in the bottom. maybe either a universal, iphone, ipad export are needed

![img_0181](https://cloud.githubusercontent.com/assets/11299322/10057201/eb29fd40-6236-11e5-981c-8b8ffff36aea.PNG)

**Comment by lewislepton** [comment]
• we need more indication of what folder we are in when clicking on a folder. maybe a simple text in the top of the hierarchy

<img width="229" alt="screen shot 2015-09-26 at 16 42 06" src="https://cloud.githubusercontent.com/assets/11299322/10118372/19c317f2-646e-11e5-967a-1f236cd2fb54.png">

**Comment by lewislepton** [comment]
• an example for changing from scene to scene

**Comment by lewislepton** [comment]
• need to have a check to stop multiple atomic players opening up. so only 1 plays at a time

<img width="519" alt="screen shot 2015-09-26 at 16 54 19" src="https://cloud.githubusercontent.com/assets/11299322/10118426/6b4fdc08-646f-11e5-9d59-02a2cc8640f7.png">

**Comment by lewislepton** [comment]
• template type loading/saving. if you have specific settings for a player/enemy/block etc with gravity, or lights or whatever, it would be good if you could save it. unlike a prefab, this would not be using the sprites, plus if saved within atomic, or a chosen path, you could reuse things

it would save a ton of time either reading, writing down settings, but you could set up a library of templates specific to what you want but are not project specific, but global to anything. plus maybe shareable to other users

the template should not be the node name, or even the sprite lodaing, it is the setting for every after that. so physics, lights and all the other components you can load in

<img width="326" alt="screen shot 2015-09-26 at 16 58 52" src="https://cloud.githubusercontent.com/assets/11299322/10118450/502aac54-6470-11e5-89c6-7df8ee463d95.png">

**Comment by lewislepton** [comment]
• IDEA FOR NEW COMPONENT
`Repeater2D` also `Repeater3D`. or just call it `Repeater` ;)
would be used for repeating assets/sprites/general world stuff. so things like platforms. it would calculate the width and height of the sprite [not physics components], then you can set how many times it repeats/copies either in the X, Y or Z co-ordinates, or all. but it should copy in the physics stuff within the node that is being copied

having a repeater could also cut out having to copy and paste assets which fill the hierarchy, and quickly get levels built especially if making the same asset in rows or slightly spaced apart

so this picture is showing that i want to repeat a sprite 23 times on the X axis, everything else is at 0. you can also space them apart by pixel width, so 32, would be 32 pixel apart from each other

<img width="312" alt="screen shot 2015-09-26 at 17 31 29" src="https://cloud.githubusercontent.com/assets/11299322/10118665/23859f96-6476-11e5-980d-d3a01f49890b.png">

**Comment by shaddockh** [comment]
Regarding template type loading/saving.

This is what the blueprint system I'm putting together is attempting to accomplish, but everything is currently being handled outside the editor.  Having something inside the editor that allows for saving/loading some common settings could be useful, or exposing some hooks to allow editor extensions that could do the same could also be very useful.

**Comment by Alan-FGR** [comment]
I don't know if this was mentioned, but when you are scrolling the hierarchy by clicking the scrollbar, if when you release the mouse it's over an object in the list, it gets selected.
Another problem related to https://github.com/AtomicGameEngine/AtomicGameEngine/issues/539 is that when you try to move multiple objects around the whole thing gets a little crazy after, that, I don't know exactly how to reproduce, but sometimes when you try to DnD multiple objects you can't DnD any longer.

**Comment by Alan-FGR** [comment]
There's another problem I have sometimes... when you open an example that uses 2D physics, you can remove the PhysicsWorld2D component but you can't add a 3D and not even the 2D one again, although I have to check the exact reproduction steps.

**Comment by Alan-FGR** [comment]
Also, a minor: when you open a menu dropdown, clicking again won't close it

**Comment by Alan-FGR** [comment]
We should maintain a tickable list of the issues reported here imho.

**Comment by Alan-FGR** [comment]
In my opinion we should also bring back the blue on/off indicator on components, like the ones on the screen shot here: https://github.com/AtomicGameEngine/AtomicGameEngine/issues/177#issuecomment-130425596 so we can easily tell whether a node/component is enabled. I think this was removed because of inconsistency, but we could go smart and add an on/off indicator on any component that has an 'enabled' variable... maybe we could use a bit of regex to catch loose matches, so the user scripts could have an 'enabled' variable that shows on the component header... just an idea

**Comment by Alan-FGR** [comment]
UPDATE: Whoah! Just realized those aren't on/off indicators but just the 'arrow' thing :P... in this case, in my opinion we really should have on/off indicators there, and maybe make it blue only when it's on, so it's easy to tell... Unity for example has that (checkbox for components that can be enabled/disabled, and nothing for the ones that can't) and it's quite handy when you have lots of components collapsed.

**Comment by Alan-FGR** [comment]
I don't know if it's best to open another issue and organize the valid points here in a list... https://github.com/AtomicGameEngine/AtomicGameEngine/issues/159#issuecomment-141226956 for example is still valid

**Comment by JoshEngebretson** [comment]
Yes, it is best to open another issue, this one is awesome, though a bit like Room 237 from "The Shining"

![shining-room-237-4001](https://cloud.githubusercontent.com/assets/376203/20736428/39fb4716-b65b-11e6-8284-73912316c6b8.jpg)

**Comment by xnakos** [comment]
@JoshEngebretson I am not sure if this is the right place to report these:

![1](https://cloud.githubusercontent.com/assets/1213128/21075028/bb42419a-bf10-11e6-969c-049e227ab632.png)

![4](https://cloud.githubusercontent.com/assets/1213128/21075051/3cb37e7e-bf11-11e6-8922-02a1a1a15be5.png)

I am on Ubuntu Desktop 16.04 64-bit and this is today's master. The scene is all over the place. I did not encounter a relevant bug report, but I may have missed something. Should I possibly open a new issue?

I tried Atomic Game Engine on Windows 7 64-bit (on the same machine) a couple of days ago and the scene was fine to work with.

**Comment by JoshEngebretson** [comment]
@xnakos Yes, please open a separate issue, thanks :)

**Comment by JoshEngebretson** [comment]
Many of the issues have been broken out as can be seen above, with some closed, there are still issues/feedback to mine out of this issue, though duplicates should be avoided.

**Comment by Alan-FGR** [comment]
@JoshEngebretson yeah, the idea is revisiting all of them and extracting from this only the ones that are still valid. Any words on multi-issues or is that dependent on whether the issue is 'big' enough to have its own entry?

**Comment by Alan-FGR** [comment]
Gonna add this on my to-do for next weekend, dunno if I'll have time to finish though. Many points are still valid. Hopefully we'll be able to close this one soon.

**Comment by Alan-FGR** [comment]
Assigning myself. Can't promise anything for soon though.

**Comment by Alan-FGR** [comment]
Irrelevant for new editor
